### PR TITLE
Mobile

### PR DESCRIPTION
**Pull request by FedericoDiTosto**
I created a new button that allows you to view ,through a new preview canvas, the next block to be inserted, change its shape, color and rotation to improve usability from mobile. 
Made the options bar responsive for mobile viewing.

### CODE REVIEW

**Comment by AlasdairMott** [comment]
that's great,  makes it so much easier on mobile!

The menu animation is a little jerky, I reckon leave it out for now. It's also on top of the help modal, as is the mobile icon. 

https://user-images.githubusercontent.com/25563790/200143874-7b9e529c-767b-4b6e-9e4b-45947445b032.mp4

I was thinking it could look something like this: 

image

On mobile the icons spill over. We can have it scale or maybe share should sit in the lower right with help and mute?

image

File: src/3d/Viewport3d.tsx [hunk 1]
```diff
@@ -152,14 +154,17 @@ const Viewport3d = () => {
...
+                <Grid />
                 <Plane receiveShadow={true} scale={100} rotation-x={-Math.PI * 0.5} position-y={-0.5}>
                     <shadowMaterial opacity={0.1} />
                 </Plane>
             </Canvas>
             <Toolbar onZoomExtents={zoomExtents} onTogglePerspective={handleTogglePerspective} onNewFile={handleNewFile}></Toolbar>
-            <HelpAndSoundButtons onShowHelp={() => setShowHelp(!showHelp)}/>
-            {showHelp && <HelpModal onClose={() => setShowHelp(false)}/>}
+            <HelpAndSoundButtons onShowHelp={() => setShowHelp(!showHelp)} />
+            {showHelp && <HelpModal onClose={() => setShowHelp(false)} />}
```
Comment (AlasdairMott) [inline src/3d/Viewport3d.tsx]: move 163 and 164 after the mobile buttons to that the help modal displays on top :)

File: src/UI/MobileSelector.module.css [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+.canvasContainer {
+    max-width: 70%;
+}
+
+.canvasContainer canvas {
+    border-radius: 12px;
+}
+
+.previwButtons {
```
Comment (AlasdairMott) [inline src/UI/MobileSelector.module.css]: typo here

File: src/UI/MobileMode.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+const MobileSelector = (props: { onClose: () => void, cam: any }) => {
+    const nextModel = useAppSelector(state => state.model.model);
+    const dispatch = useDispatch();
+    return (
+        <div className={styles.mobileSelector}>
+            <div className={styles.canvasContainer}>
+                <Canvas shadows={true} camera={{ position: [1, 1, 1] }}>
+                    <color attach="background" args={[1, 1, 1]} />
+                    <OrbitControls />
```
Comment (AlasdairMott) [inline src/UI/MobileMode.tsx]: not sure if the preview will need orbit controls, adds some complexity if the user pans or zooms out far

File: src/UI/MobileMode.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                    <ambientLight intensity={1.0} />
+                    <Model {...nextModel} />
+                </Canvas>
+            </div>
+            <div className={`${styles.previwButtons} glass`}>
+                <ToolbarCommandButton onClick={() => { console.log(props.cam); dispatch(modelActions.nextBlock()) }} title={"next-shape"}><AddHexagon {...ICONPROPS} />
+                </ToolbarCommandButton>
+                <ToolbarCommandButton onClick={() => { dispatch(modelActions.rotateBlock('right')) }} title={"change-color"}><CropRotateTr {...ICONPROPS} />
```
Comment (AlasdairMott) [inline src/UI/MobileMode.tsx]: Its not ideal but I reckon iconoir`refresh-double` would be clearer here

**Comment by FedericoDiTosto** [comment]
Great idea, the menu is much cleaner this way. I will fix it as soon as I can. 
I tried to make the mobile menu canvas rotate every time you move with the OrbitControl the main scene but without success(so that the user knows exactly at what angle the new block will be positioned). Do you know how to make two canvas rotate with only one OrbitControl?

**Comment by FedericoDiTosto** [comment]
I've fixed the layout, let me know if I've forgotten anything or you have new ideas. For the top toolbar I fixed (only for some devices) by decreasing the side margin obviously it is a temporary solution.

**Comment by AlasdairMott** [comment]
Wow that looks so good! Looks great on my mobile.

> Do you know how to make two canvas rotate with only one OrbitControl?

I tried doing something like this to keep the perspective and ortho cameras in sync, but in the end I had to only sync when swapping over. 

A workaround might be an xyz gizmo in the viewport + preview.

**Pull request merged**